### ISSUE TITLE
Publishing manylinux wheel seems to have been broken for a few releases

### ISSUE THREAD

**Issue by Andrewpk** [pola-rs/polars#13799]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
echo "polars==0.20.5" > requirements.txt
pip install -r requirements.txt -t deps --upgrade --platform manylinux2014_aarch64 --only-binary=:all: --no-compile
```

### Log output

```shell
ERROR: Ignored the following yanked versions: 0.14.12, 0.14.13, 0.14.15, 0.14.20, 0.14.31, 0.15.0, 0.15.3, 0.15.4, 0.15.5, 0.15.10, 0.16.17, 0.17.0, 0.17.1, 0.17.2, 0.17.6, 0.17.7, 0.17.8, 0.18.1, 0.18.5, 0.18.9, 0.18.10, 0.18.14, 0.19.4, 0.19.8, 0.19.9, 0.19.10
ERROR: Could not find a version that satisfies the requirement polars==0.20.5 (from versions: 0.14.8, 0.14.9, 0.14.10, 0.14.14, 0.14.17, 0.14.18, 0.14.19, 0.14.21, 0.14.23, 0.14.24, 0.14.25, 0.14.26, 0.14.27, 0.14.28, 0.14.29, 0.15.1, 0.15.2, 0.15.6, 0.15.7, 0.15.8, 0.15.9, 0.15.11, 0.15.13, 0.15.14, 0.15.15, 0.15.16, 0.15.17, 0.15.18, 0.16.1, 0.16.2, 0.16.3, 0.16.4, 0.16.5, 0.16.6, 0.16.7, 0.16.8, 0.16.9, 0.16.10, 0.16.11, 0.16.12, 0.16.13, 0.16.14, 0.16.15, 0.16.16, 0.16.18, 0.17.5, 0.17.9, 0.17.10, 0.17.11, 0.17.12, 0.17.13, 0.17.14, 0.17.15, 0.18.0, 0.18.2, 0.18.3, 0.18.4, 0.18.6, 0.18.7, 0.18.8, 0.18.11, 0.18.12, 0.18.13, 0.18.15, 0.19.0, 0.19.1, 0.19.2, 0.19.3, 0.19.5, 0.19.6, 0.19.7, 0.19.11, 0.19.12rc1, 0.19.12)
ERROR: No matching distribution found for polars==0.20.5
```

### Issue description

It appears `0.19.12` is the last release to have a wheel published for `manylinux2014_aarch64`. I saw another issue that looked like this had been resolved (and looking at the releases available, it looks like it was for some time), but I'm wondering if something broke somewhere.

I'm not actually sure where the release build for this was done as I couldn't find it in actions but I'm also wholly unfamiliar with building wheels myself. If you give me some hints where to look I can always try to drop an MR in.

### Expected behavior

I'd expect wheels to be built for manylinux2014_aarch64. When working on AWS Lambdas that are running on arm64 this is mostly a requirement in order to use the dependency.

### Installed versions

<details>

```
adbc_driver_sqlite:  <not installed>
cloudpickle:         <not installed>
connectorx:          <not installed>
deltalake:           <not installed>
fsspec:              <not installed>
gevent:              <not installed>
matplotlib:          <not installed>
numpy:               <not installed>
openpyxl:            <not installed>
pandas:              <not installed>
pyarrow:             <not installed>
pydantic:            1.10.13
pyiceberg:           <not installed>
pyxlsb:              <not installed>
sqlalchemy:          <not installed>
xlsx2csv:            <not installed>
xlsxwriter:          <not installed>

```

</details>

**Comment by s-banach** [comment]
manylinux2014 is too old, will not be supported again. It’s officially end of life this year anyway.

**Comment by Andrewpk** [comment]
TIL about manylinux_2_24 and   manylinux_2_28. Sorry for the erroneous issue.

**Comment by Andrewpk** [comment]
I may have closed this issue too soon @s-banach. It appears all of the current AWS docs refer to manylinux2014 still and many common deps still don't have manylinux_2_24 or 2_28 wheels out yet :(
manylinux2014 seems to be the only one to fulfill a simple requirement of:
```
python = "^3.11"
requests = "^2.31"
pycognito = "^2023.5"
fastapi = "^0.109"
mangum = "^0.17.0"
boto3-stubs = {extras = ["s3", "secretsmanager", "dynamodb"], version = "^1.34"}
aws-lambda-powertools = {extras = ["parser", "tracer", "validation"], version = "^2.31.0"}
polars = "0.19.12"
```
I think I'm going to concede that working with dependencies and AWS Lambdas still is a terrible experience and just use a docker image to build the dependency layer so I can at least get a current polars.

**Comment by Andrewpk** [comment]
Using my docker container based off `public.ecr.aws/sam/build-python3.11:latest-arm64` to install the deps for my lambda layer worked great. I was able to pull in polars-0.20.5-cp38-abi3-manylinux_2_24_aarch64.whl this way.

tl;dr: just use a docker container to install your python package dependencies for lambdas.
